File: 0013-roman-to-integer/0013-roman-to-integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        int alpha[28] = {0};
        vector<int> converted;
        int answer=0;
        alpha['I'-'A'] = 1;
        alpha['V'-'A'] = 5;
        alpha['X'-'A'] = 10;
        alpha['L'-'A'] = 50;
        alpha['C'-'A'] = 100;
        alpha['D'-'A'] = 500;
        alpha['M'-'A'] = 1000;
        
        for(char c:s){
            converted.push_back(alpha[c-'A']);
        }
        
        int latest=0;
        for(int i=converted.size()-1; i>=0; i--){
            if(converted[i]>=latest) answer+=converted[i];
            else answer-=converted[i];
            latest=converted[i];
        }
        
        return answer;
    }
};

File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        n = len(s)
        numMap = {'I':1, 'V':5, 'X':10, 'L':50, 'C':100, 'D':500, 'M':1000}
        answer=0
        for i in range(n-1):
            #print(i)
            if numMap[s[i]]>=numMap[s[i+1]]:
                answer+=numMap[s[i]]
            else:
                answer-=numMap[s[i]]
        answer+=numMap[s[n-1]]
        
        return answer